test/e2e_node/device_plugin.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
+	dp "k8s.io/kubernetes/pkg/kubelet/cm/deviceplugin"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Serial because the test restarts Kubelet
+var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin] [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("device-plugin-errors")
+
+	Context("DevicePlugin", func() {
+		By("Enabling support for Device Plugin")
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FeatureGates[string(features.DevicePlugins)] = true
+		})
+
+		It("Verifies the Kubelet device plugin functionality.", func() {
```

Comment:
Is it possible that we define a few helper functions in this file: setupLocal (creates device files and mounted directories. called from BeforeEach()), cleanupLocal(removes any created files and mount directories. called from AfterEach()), allocFunc (creates AllocateResponse with some test DeviceSpec, Mounts, and Envs. for simplicity, we can use const mount points and envs that are easier to verify in user pod), and podRECMD(returns a cmd string that allows us to verify test DeviceSpec, Mounts, and Envs in the user pod).

Reply:
I add a allocFunc() in stub_device_plugin as you suggested above :)

Reply:
And I think put these creation/deletion together in `allocFunc()` maybe better.
This PR is a basic test without verifying `DeviceSpec`, `Mounts`, and `Envs`, maybe be we can extend this podRECMD later.

Reply:
Sounds a reasonable plan to verify container runtime setting in a follow-up pr.